### PR TITLE
Multiple samples - Add done loading listeners to avoid silent failure

### PR DESCRIPTION
**Pull request by JonLavi**
We've recently had a lot of customers pointing out troubles with loadables, which is why we need to highlight this pattern a bit more. I'm also going to piggyback the issue regarding garbage collection and make some of the changes here where I've already had the sample open.

* Made some of the loadables class members, to avoid garbage collection

* Added checks for LoadStatus.LOADED, either showing a message if failed to load, or enabling the rest of the sample's logic.
	Note that we're triggering the layer's loading process by adding it to the map _outside_ of the doneLoadingListener, and just using the latter for error reporting or enabling functionality. The alternative would have been to add the layers to the map inside the listener, and start the loading with `loadAsync()`. However that can be a bit confusing and shouldn't be the pattern we promote. 

* in a couple of samples I've removed service URL class members, and just put the URLs in-line when the appropriate table/layer is created

cc @mbcoder

### CODE REVIEW

File: display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java
Comment (Rachael-E) [inline display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java]: just a heads up from https://github.com/Esri/arcgis-runtime-samples-java/pull/511/files which is still to be approved: when putting loadables as member variables to protect against garbage collection, I was adding the following comment to make it clear why we did this. This comment may or may not be agreed on but wanted to doc it here if it's approved so we can add the same comment here for the loadables.
```suggestion
// keeps loadables in scope to avoid garbage collection
    private FeatureLayer riverFeatureLayer;
```

File: display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java
Comment (Rachael-E) [inline display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java]: I'm thinking a slight rewording of the comments here and throughout the PR, what do you think?
```suggestion
            // check that the feature layer has loaded by adding a done loading listener (triggered asynchronously)
            // check for the load status of the layer and if it hasn't loaded, report an error
```

File: image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java
Comment (Rachael-E) [inline image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java]: we can "plain english" this comment a little more
```suggestion
      // create an ArcGISMap with the topographic basemap and set it to the map view
```

**Review by Rachael-E — APPROVED** [review]
thanks @JonLavi looks good. I've put in some comment suggestions, take or leave them as you like!

File: display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java
Comment (JonLavi) [reply display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java]: Definitely need to add the same comment in these samples! 👍

File: display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java
Comment (JonLavi) [reply display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java]: I'm having some problems with this wording, because to me it reads as if the action of adding the listener is what's triggered asynchronously. Also the repetition of 'check' in the beginning bugs me a bit. How about:
```suggestion
            // add a done loading listener, with a runnable that is triggered asynchronously when the feature layer has loaded
            // check for the load status of the layer and if it hasn't loaded, report an error
```
A bit wordy, but hopefully clearer?

**Comment by JonLavi** [comment]
Thanks for the comments @Rachael-E ! I've made the Layers that are added to OperationalLayers local variables again, following Colins advice in your PR, that these won't get garbage collected. Let me know what you think about the reworded comment for the done loading listener?

File: display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java
Comment (Rachael-E) [reply display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java]: @JonLavi yeh I like this wording! Minor change I'd make would be to replace "gets" with "is", just reads a little better I find.

File: editing/update-attributes/src/main/java/com/esri/samples/update_attributes/UpdateAttributesSample.java
Comment (Rachael-E) [inline editing/update-attributes/src/main/java/com/esri/samples/update_attributes/UpdateAttributesSample.java]: with your comment from my garbage collection PR in mind, should we change this to `selectedFeature`?

**Review by Rachael-E — APPROVED** [review]
thanks @JonLavi, looks good! I left a comment in response to re-wording and a note here about changing variable name from `selected` to `selectedFeature`.

File: editing/update-attributes/src/main/java/com/esri/samples/update_attributes/UpdateAttributesSample.java
Comment (JonLavi) [reply editing/update-attributes/src/main/java/com/esri/samples/update_attributes/UpdateAttributesSample.java]: Definitely! Thanks for pointing that out!

**Comment by JonLavi** [comment]
Thanks for the review @Rachael-E !
@tschie could you review when you have some time? Thanks!

File: display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java
Comment (tschie) [inline display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java]: too verbose

```suggestion
            // show alert if layer fails to load
```

File: display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java
Comment (tschie) [inline display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java:72]: This is okay, but seems a little overkill for a simple sample. We've had discussions before about omitting some best practices for the sake of brevity/focus. Should get approval from Lucas/Mike before making more of these changes.

File: display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java
Comment (tschie) [inline display_information/display-annotation/src/main/java/com/esri/samples/display_annotation/DisplayAnnotationSample.java]:
```suggestion
            annotationLayer.addDoneLoadingListener(() -> {
```

File: editing/update-attributes/src/main/java/com/esri/samples/update_attributes/UpdateAttributesSample.java
Comment (tschie) [inline editing/update-attributes/src/main/java/com/esri/samples/update_attributes/UpdateAttributesSample.java]:
```suggestion
      // create a feature layer with the feature table
```

File: editing/update-attributes/src/main/java/com/esri/samples/update_attributes/UpdateAttributesSample.java
Comment (tschie) [inline editing/update-attributes/src/main/java/com/esri/samples/update_attributes/UpdateAttributesSample.java]:
```suggestion
      // add the feature layer to the map
```

File: editing/update-attributes/src/main/java/com/esri/samples/update_attributes/UpdateAttributesSample.java
Comment (tschie) [inline editing/update-attributes/src/main/java/com/esri/samples/update_attributes/UpdateAttributesSample.java]: too verbose

File: editing/update-attributes/src/main/java/com/esri/samples/update_attributes/UpdateAttributesSample.java
Comment (tschie) [inline editing/update-attributes/src/main/java/com/esri/samples/update_attributes/UpdateAttributesSample.java]: Perhaps `identifiedFeature` would be a better name? Looks odd to select the selected feature.

File: image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java
Comment (tschie) [inline image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java]: I realize there wasn't a comment before, but we should really have one for why we changed the opacity

File: image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java
Comment (tschie) [inline image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java:100]: if only we had a visible property to bind to.... @mbcoder 😜

File: raster/raster-rendering-rule/src/main/java/com/esri/samples/raster_rendering_rule/RasterRenderingRuleSample.java
Comment (tschie) [inline raster/raster-rendering-rule/src/main/java/com/esri/samples/raster_rendering_rule/RasterRenderingRuleSample.java]: too verbose

**Review by tschie — APPROVED** [review]
Looks good. Just a few minor things

**Pull request merged**
